collex-loader/load.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+import argparse
+import csv
+from functools import partial
+import json
+
+import requests
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Load collection exercise CSV.')
+    parser.add_argument("config", help="Configuration file")
+    return parser.parse_args() 
+
+
+def map_columns(column_mappings, row):
+    new_row = dict()
+    for key, value in row.items():
+        if key and value:
+            new_row[column_mappings[key] if column_mappings[key] else key] = value
+    return new_row
+
+
+def post_collex(data, url, user, password):
+    response = requests.post(url, json=data, auth=(user, password))
+
+    status_code = response.status_code
+    detail_text = response.text if status_code != 201 else ''
+
+    print("%s <= %s (%s)" % (status_code, data, detail_text))
+
+
+def dump_collex(data):
+    survey_id = data['surveyRef']
+    period = data['exerciseRef']
+    if survey_id and period:
+        filename = "%s-%s.json" % (survey_id, period)
+        with open(filename, 'w') as fo:
+            json.dump(data, fo)
+
+
+def clean_row(row):
+    row['name'] = row['name'][:20]
+
+    return row
+        
+
+def process_file(filename, row_handler):
+    with open(filename) as fp:
+        reader = csv.DictReader(fp)
+        for row in reader:
+            new_row = map_columns(column_mappings, row)
+
+            if new_row:
+                row_handler(data=clean_row(new_row))
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    print("Config filename: %s" % args.config)
```

Comment:
Is there a reason these are being printed rather than logged?

Reply:
It's a stand-alone script. So I don't think we need to worry about setting up a logger.